lib/puppet/provider/package/rpm.rb

```diff
@@ -281,6 +281,7 @@ def self.nevra_to_hash(line)
       self::NEVRA_FIELDS.zip(match.captures) { |f, v| hash[f] = v }
       hash[:provider] = self.name
       hash[:ensure] = "#{hash[:version]}-#{hash[:release]}"
+      hash[:ensure].prepend("#{hash[:epoch]}:") if hash[:epoch] != '0'
```

Comment:
@joshcooper hm, it looks like the only places the RPM provider uses `versioncmp` is to check the version of RPM itself though, right?

From what I recall, the package that's actually being installed gets it's ensure version compared in the yum provider,  at this method: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/provider/package/yum.rb#L226

From my previous (maybe flawed) understanding, it's the child providers of `rpm` that deal with the epoch. `rug`, for example, says: `# XXX: We don't actually deal with epochs here.` in it's `install` method. `zypper` and `urpmi` also do this. `Yum` seems to be the only provider that does any extensive version comparisons at all, for some reason.

(Also, there actually already is an `rpmvercmp` method, also, which is pretty gnarly: https://github.com/whopper/puppet/blob/PUP-5025/RPM-epoch/lib/puppet/provider/package/rpm.rb#L182)